packages/roosterjs-editor-core/lib/corePlugins/DOMEventPlugin.ts

```diff
@@ -152,8 +152,18 @@ export default class DOMEventPlugin implements PluginWithState<DOMEventPluginSta
 
     private onContextMenuEvent = (event: MouseEvent) => {
         const allItems: any[] = [];
+        const position = this.editor.getFocusedPosition();
+        let eventTargetNode = event.target as Node;
+        if (
+            position &&
+            event.button != 2 &&
+            safeInstanceOf(position.node, 'HTMLDivElement') &&
```

Comment:
why check DIV?

I think you can get InlineElement before cursor and use its container node as the node. 

Use editor.getContentSearcherOfCursor(event), then call searcher.getInlineElementBefore().

It doesn't need to be image, but for whatever inline element we can use it.